src/main/java/org/mitre/synthea/world/agents/Person.java

```diff
@@ -240,6 +241,24 @@ public Period age(long time) {
     return age;
   }
 
+  /**
+   * Returns a person's age in decimal years. (ex. 7.5 ~ 7 years 6 months old)
+   *
+   * @param time The time when their age should be calculated.
+   * @return decimal age in years
+   */
+  public double ageInDecimalYears(long time) {
+    double age = 0.0;
+
+    if (attributes.containsKey(BIRTHDATE)) {
+      LocalDate now = Instant.ofEpochMilli(time).atZone(ZoneId.systemDefault()).toLocalDate();
+      LocalDate birthdate = Instant.ofEpochMilli((long) attributes.get(BIRTHDATE))
+          .atZone(ZoneId.systemDefault()).toLocalDate();
+      age = birthdate.until(now, IsoFields.QUARTER_YEARS) / 4.0;
+    }
+    return age;
+  }
```

Comment:
It seems like you are unnecessarily copying code... couldn't you just do something like:

```java
double years = (double) age(time).toTotalMonths() / 12.0;
if (years < 0) {
  years = 0;
}
return years;
```